### PR TITLE
Update mail helper v3 with feedback

### PR DESCRIPTION
**Pull request by GustavoCaso**
#177 

@thinkingserious I added some of the feedback, I'm sure leaving some behind but I wanted to know if so far you are ok with the changes.

Also I have seen methods like `add_header` or `add_headers` maybe we could simplify by just having one method. What do you think?

I will wait for your feedback, so we can start coding.

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=190) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/sendgrid-ruby?pullRequest=190) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/sendgrid-ruby?pullRequest=190) it.</sub>

**Comment by mbernier** [comment]
This looks like it might not be backward compatible if we change the class names. Is it possible to do some magic to make the old ClientFactory class still work? Maybe if ClientFactory became a wrapper on Client or vice-versa?

File: mail_helper_v3.md [hunk 1]
```diff
@@ -116,33 +122,33 @@ puts response.headers
...
 plain_text_content = 'and easy to do anywhere, even with Ruby'
 html_content = '<strong>and easy to do anywhere, even with Ruby</strong>'
-msg = SendGrid::SendGridMessage.new(from, to, subject, plain_text_content, html_content)
+msg = SendGrid::Message.new(from, to, subject, plain_text_content, html_content)
 
 # For a detailed description of each of these settings, please see the [documentation](https://sendgrid.com/docs/API_Reference/api_v3.html).
 
-msg.add_to(SendGrid::Email.new('test1@example.com', 'Example User1'))
-to_emails = [ 
+msg.add_recipient(SendGrid::Email.new('test1@example.com', 'Example User1'))
```
Comment (thinkingserious) [inline mail_helper_v3.md]: Let's please stay with `add_to`

File: mail_helper_v3.md [hunk 2]
```diff
@@ -116,33 +122,33 @@ puts response.headers
...
+to_emails = [
     SendGrid::Email.new('test2@example.com', 'Example User2'),
     SendGrid::Email.new('test3@example.com', 'Example User3')
 ];
-msg.add_tos(to_emails)
+msg.add_recipients(to_emails)
```
Comment (thinkingserious) [inline mail_helper_v3.md]: Let's please stay with `add_tos`

**Comment by GustavoCaso** [comment]
@thinkingserious I update with your requests. Is there anything else you need to finish the `v3` specs?

**Comment by thinkingserious** [comment]
Hello @GustavoCaso,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/announcing-hacktoberfest-2017">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/swag">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Pull request merged**